com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -0,0 +1,1095 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Profiling;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// The NGO connection manager handles:
+    /// - Client Connections
+    /// - Client Approval
+    /// - Processing <see cref="NetworkEvent"/>s.
+    /// - Client Disconnection
+    /// - MessagingSystem updates
+    /// </summary>
+    // TODO 2023-Q2: Discuss what kind of public API exposure we want for this
+    public class NetworkConnectionManager : INetworkUpdateSystem
+    {
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+        private static ProfilerMarker s_TransportPollMarker = new ProfilerMarker($"{nameof(NetworkManager)}.TransportPoll");
+        private static ProfilerMarker s_TransportConnect = new ProfilerMarker($"{nameof(NetworkManager)}.TransportConnect");
+        private static ProfilerMarker s_HandleIncomingData = new ProfilerMarker($"{nameof(NetworkManager)}.{nameof(MessagingSystem.HandleIncomingData)}");
+        private static ProfilerMarker s_TransportDisconnect = new ProfilerMarker($"{nameof(NetworkManager)}.TransportDisconnect");
+#endif
+
+        /// <summary>
+        /// The current host name we are connected to, used to validate certificate
+        /// </summary>
+        public string ConnectedHostname { get; private set; }
```

Comment:
I should have been clearer in my comment. I was actually recommending returning an empty string from the getter in `NetworkManager`, and simply not having the property in `NetworkConnectionManager`. I get that the property needs to stay in `NetworkManager` to avoid breaking the API, but it doesn't have to be in new APIs like `NetworkConnectionManager`, as this makes it look like the API is actually useful.

Reply:
done as you suggested, see c8dd262